src/compiler/checker.ts

```diff
@@ -38913,7 +38922,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         // Check if the index type is assignable to 'keyof T' for the object type.
         const objectType = (type as IndexedAccessType).objectType;
         const indexType = (type as IndexedAccessType).indexType;
-        if (isTypeAssignableTo(indexType, getIndexType(objectType, IndexFlags.None))) {
+        if (isTypeAssignableTo(indexType, getIndexType(objectType, IndexFlags.NoRemappingMappedTypeDeferral))) {

```

Comment:
I kinda expected that I would have to pass this at more locations. Given the fact that this is the only call site using this... maybe I just should remove this `IndexFlags.NoRemappingMappedTypeDeferral` and perform the targeted simplification of `keyof RemappingMappedType` here to pass it to the `isTypeAssignableTo`.

Reply:
That would probably be better - `IndexFlags` originally only existed to support the `keyofStringsOnly` flag, and its' other uses have kind of become a convenience. With `keyofStringsOnly` likely to go away soonish, it'd be nice if we could stop having all these quiet variant index types.

Reply:
pushed out this change